### PR TITLE
Provide safeguards during training

### PR DESCRIPTION
**Pull request by RobotSail**
This PR adds safeguards to prevent the following issues:

- preventing the multipack sampler from being used when it cannot adequately process all samples
- prevent empty datasets from being provided or created
- prevent `max_batch_len` from being smaller than `max_seq_len`

### CODE REVIEW

**Review by Maxusmusti — APPROVED** [review]
LGTM

File: src/instructlab/training/data_process.py [hunk 1]
```diff
@@ -205,6 +205,10 @@ def main(args: DataProcessArgs):
     )
 
     data = load_dataset("json", data_files=args.data_path, split="train")
+    if data.num_rows == 0:
+        raise ValueError(
+            "The provided dataset is empty, please make sure that you were able to generate a dataset and try again."
```
Comment (JamesKunstle) [inline src/instructlab/training/data_process.py:210]: I'd make this value error generation-agnostic.

File: src/instructlab/training/main_ds.py [hunk 2]
```diff
@@ -555,6 +555,25 @@ def main(args):
         sampler=args.sampler,
         seed=args.seed,
     )
+    if len(train_loader) == 0:
```
Comment (JamesKunstle) [inline src/instructlab/training/main_ds.py:558]: this makes training slower so maybe we shouldn't automatically switch over to the distributed data sampler.

Comment (JamesKunstle) [reply src/instructlab/training/main_ds.py:558]: @Maxusmusti and I talked about it- defaulting to naive distribution seems reasonable

File: src/instructlab/training/data_process.py (hunk 1 above)
Comment (RobotSail) [reply src/instructlab/training/data_process.py:210]: Good call, changing this

**Review by Maxusmusti — APPROVED** [review]
Tested myself as well

**Pull request merged**